synmark/suites/logging.py

```diff
@@ -11,60 +11,63 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
 import logging
+import logging.config
 import warnings
 from io import StringIO
+from typing import Optional
 from unittest.mock import Mock
 
 from pyperf import perf_counter
 
+from twisted.internet.address import IPv4Address, IPv6Address
 from twisted.internet.defer import Deferred
 from twisted.internet.protocol import ServerFactory
 from twisted.logger import LogBeginner, LogPublisher
 from twisted.protocols.basic import LineOnlyReceiver
 
 from synapse.config.logger import _setup_stdlib_logging
 from synapse.logging import RemoteHandler
+from synapse.synapse_rust import reset_logging_config
+from synapse.types import ISynapseReactor
 from synapse.util import Clock
 
 
 class LineCounter(LineOnlyReceiver):
     delimiter = b"\n"
+    count = 0
```

Comment:
Is this safe? `count` is now a class attribute. If something else modifies `self.count`, does it overwrite the class attribute or make a new instance attribute?

Reply:
It isn't mutable so this is safe:

```pycon
>>> class A:
...   count = 0
... 
>>> a = A()
>>> b = A()
>>> c = A()
>>> a.count, b.count, c.count
(0, 0, 0)
>>> a.count = 3
>>> a.count, b.count, c.count
(3, 0, 0)
>>> c.count += 1
>>> a.count, b.count, c.count
(3, 0, 1)
```